pkg/api/v1/router_instance_metadata.go

```diff
@@ -275,6 +282,69 @@ func (r *Router) instanceMetadataSet(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+// Log the difference between the metadata we're about to upsert and the metadata we have in the database
+func instanceMetadataLogDiffs(r *Router, c *gin.Context, newInstanceMetadata *models.InstanceMetadatum) {
+	instanceID := newInstanceMetadata.ID
+
+	existingMetadata, err := models.FindInstanceMetadatum(c.Request.Context(), r.DB, instanceID)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		r.Logger.Sugar().Info("Upsert metadata diff for instance ", instanceID, ": this is the first metadata for this instance (sql.ErrNoRows)")
```

Comment:
I suspect this is going to be *really* noisy and not very helpful to have to weed out, doesn't feel necessary to log this condition.